### PR TITLE
Fixed PHP-1425: explain() does not raise appropriate exception for  conditions

### PR DESCRIPTION
**Pull request by derickr**

### CODE REVIEW

File: tests/generic/bug01425.phpt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+}
+catch ( MongoCursorException $e )
+{
+	echo $e->getCode(), "\n";
+
+	$m->admin->command( array( 'setParameter' => 1, 'notablescan' => 0 ) );
+}
+
+$m->admin->command( array( 'setParameter' => 1, 'notablescan' => 0 ) );
+?>
```
Comment (bjori) [inline tests/generic/bug01425.phpt]: It doesn't have to be both in the catch() and outside it.
Also, on failure this test won't print anything - thats abit confusing. how about ending with echo ok or ===DONE=== ? :]

**Comment by bjori** [comment]
Looks like travis doesn't agree with you on the error code.

Was explain() the only helper that relied on MongoCursor::next() doing the error handling? No other helpers missing php_mongo_handle_error() ?

File: tests/generic/bug01425.phpt (hunk 1 above)
Comment (derickr) [inline tests/generic/bug01425.phpt]: I know it doesn't need to be in both, but it's just to be on the safe side :-)

**Comment by jmikola** [comment]
> Was explain() the only helper that relied on MongoCursor::next() doing the error handling? No other helpers missing php_mongo_handle_error() ?

From a quick grep, it looks like the legacy collection/index enumeration, `php_mongo_runcommand`, and `php_mongo_collection_findone` all called `php_mongo_handle_error`, as did the relevant cursor functions. Of note, `findAndModify` uses `php_mongo_runcommand`, so we're covered there. I can't imagine the other command helpers do anything differently.

**Comment by bjori** [comment]
lgtm after then.
the new testcase also reads better \o/

**Pull request merged**